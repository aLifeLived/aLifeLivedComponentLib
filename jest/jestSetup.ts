/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
