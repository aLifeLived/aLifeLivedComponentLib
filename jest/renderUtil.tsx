import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
