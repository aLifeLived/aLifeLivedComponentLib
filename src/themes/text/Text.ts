import { createText, ResponsiveValue } from '@shopify/restyle';
import { ColorThemeTokens } from '../palette';
import { Theme } from '../theme';
import { ValueOf } from '../../ts/utils';

const FontFamilies = {
  sfUiTextBold: 'SFUIText-Bold',
  sfUiTextRegular: 'SFUIText-Regular',
  sfUiTextSemiBold: 'SFUIText-Semibold',
  sfUiDisplayRegular: 'SFUIDisplay-Regular',
};

type TextVariantToken =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'intro'
  | 'body'
  | 'bodySmall'
  | 'bodyXSmall';

type TextVariant = {
  fontFamily: ValueOf<typeof FontFamilies>;
  fontSize: number | ResponsiveValue<number, Theme>;
  lineHeight?: number | string;
  color: ColorThemeTokens;
};

export const textVariants: Record<TextVariantToken, TextVariant> = {
  h1: {
    fontSize: 52,
    lineHeight: 52,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h2: {
    fontSize: 32,
    lineHeight: 40,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h3: {
    fontSize: 24,
    lineHeight: 28.8,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h4: {
    fontSize: 20,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextRegular,
  },
  intro: {
    fontSize: 16,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextSemiBold,
  },
  body: {
    fontSize: 18,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodySmall: {
    fontSize: 16,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodyXSmall: {
    fontSize: 14,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
};

export const Text = createText<Theme>();
