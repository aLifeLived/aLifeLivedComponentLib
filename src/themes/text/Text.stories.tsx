import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { Text } from './Text';
import { CenterView } from '../../components/centerView/CenterView';

storiesOf('Text', module)
  .addDecorator(Story => <CenterView children={Story()} />)
  .add('Text', () => (
    <>
      <Text variant="h1">H1 Heading</Text>
      <Text variant="h2">H2 Heading</Text>
      <Text variant="h3">H3 Heading</Text>
      <Text variant="h4">H4 Heading</Text>
      <Text variant="body">Body</Text>
    </>
  ));
