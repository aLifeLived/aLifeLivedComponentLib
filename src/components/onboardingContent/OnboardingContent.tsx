import React, { ReactChild } from 'react';
import { Pressable } from 'react-native';

// Components
import { Box } from '../../themes/box/Box';
import { Text } from '../../themes/text/Text';

type OnboardingProps = {
  title: string;
  description: string;
  onNextPress: () => void;
  buttonTestID?: string;
  textTestID?: string;
  onNextButtonText: 'Tap next' | 'Finish';
  children: ReactChild;
};

export const OnboardingContent: React.FC<OnboardingProps> = ({
  title,
  description,
  onNextPress,
  buttonTestID,
  textTestID,
  onNextButtonText,
  children,
}) => {
  return (
    <Box
      flex={1}
      justifyContent="space-evenly"
      alignContent="center"
      alignItems="center"
    >
      <Box width="100%" alignItems="center" justifyContent="center" flex={2}>
        {children}
      </Box>

      <Box alignItems="center" flex={1} width="100%">
        <Text variant="h3" marginTop="xl" testID={textTestID}>
          {title}
        </Text>
        <Text
          variant="bodyXSmall"
          marginTop="m"
          paddingHorizontal="s"
          textAlign="center"
        >
          {description}
        </Text>
        <Box alignSelf="flex-end">
          <Pressable onPress={() => onNextPress()} testID={buttonTestID}>
            <Text variant="bodyXSmall" fontStyle="italic" padding="xl">
              {onNextButtonText}
            </Text>
          </Pressable>
        </Box>
      </Box>
    </Box>
  );
};
