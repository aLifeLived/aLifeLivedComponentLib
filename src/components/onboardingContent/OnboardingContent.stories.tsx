import React from 'react';
import { Dimensions } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';

// Component
import { OnboardingContent } from './OnboardingContent';
import { SystemFlatList } from '../../themes/systemFlatList/SystemFlatList';

// UI
import { Box } from '../../themes/box/Box';
import { Text } from '../../themes/text/Text';

const pages = [
  {
    title: 'Listen',
    description: 'This is the first description',
    onNextPress: () => action('Tap next pressed'),
    children: <Text variant="h1">This is the first</Text>,
    onNextButtonText: 'Tap next',
  },
  {
    title: 'Record',
    description: 'This is the second description',
    onNextPress: () => action('Tap next pressed'),
    children: <Text variant="h1">This is the second</Text>,
    onNextButtonText: 'Finish',
  },
];

storiesOf('Onboarding', module)
  .addDecorator(Story => Story())
  .add('Default', () => {
    const window = Dimensions.get('screen');
    return (
      <Box width={window.width} height={window.height}>
        <OnboardingContent
          title={text('Title', 'Listen')}
          description={text('Description', 'This is the description')}
          onNextPress={action('Tap next pressed')}
          onNextButtonText="Tap next"
        >
          <Text variant="h1">I'm the child</Text>
        </OnboardingContent>
      </Box>
    );
  })
  .add('Slider', () => {
    const window = Dimensions.get('screen');
    return (
      <SystemFlatList
        width={window.width}
        height={window.height}
        data={pages}
        showsHorizontalScrollIndicator={false}
        bounces={false}
        pagingEnabled
        horizontal
        keyExtractor={item => item.title}
        initialNumToRender={1}
        renderItem={({ item }) => (
          <Box width={window.width} height={window.height}>
            <OnboardingContent
              title={item.title}
              description={item.description}
              onNextPress={item.onNextPress}
              onNextButtonText={item.onNextButtonText}
            >
              {item.children}
            </OnboardingContent>
          </Box>
        )}
      />
    );
  });
