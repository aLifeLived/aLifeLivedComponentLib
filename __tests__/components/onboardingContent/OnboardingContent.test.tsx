import React from 'react';
import { render, fireEvent } from '../../../jest/renderUtil';

// Component
import { OnboardingContent } from '../../../src/components/onboardingContent/OnboardingContent';

// UI
import { Text } from '../../../src/themes/text/Text';

const mockOnNextButtonPress = jest.fn();

const setup = () =>
  render(
    <OnboardingContent
      title="Test title"
      description="This is a test description"
      onNextPress={mockOnNextButtonPress}
      testID="testId"
      onNextButtonText="Tap next"
    >
      <Text>Test child</Text>
    </OnboardingContent>
  );

describe('components >> onBoardingContent', () => {
  test('Should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should find title correctly', () => {
    const { getByText } = setup();
    expect(getByText('Test title')).toBeDefined();
  });

  test('Should find description correctly', () => {
    const { getByText } = setup();
    expect(getByText('This is a test description')).toBeDefined();
  });

  test('Should press on next button by test id', () => {
    const { getByTestId } = setup();
    fireEvent.press(getByTestId('testId'));
    expect(mockOnNextButtonPress).toBeCalledTimes(1);
  });

  test('Should render on next button test', () => {
    const { getByText } = setup();
    expect(getByText('Tap next')).toBeDefined();
  });

  test('Should render child correctly', () => {
    const { getByText } = setup();
    expect(getByText('Test child')).toBeDefined();
  });
});
