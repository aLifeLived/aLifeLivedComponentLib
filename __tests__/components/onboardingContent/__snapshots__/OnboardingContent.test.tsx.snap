// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> onBoardingContent Should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "justifyContent": "space-evenly",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 2,
          "justifyContent": "center",
          "width": "100%",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
          },
        ]
      }
    >
      Test child
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "width": "100%",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontFamily": "SFUIText-Bold",
            "fontSize": 32,
            "lineHeight": 40,
            "marginTop": 40,
          },
        ]
      }
    >
      Test title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontFamily": "SFUIDisplay-Regular",
            "fontSize": 18,
            "lineHeight": 20,
            "marginTop": 16,
            "paddingHorizontal": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      This is a test description
    </Text>
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-end",
            "display": "flex",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="testId"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#282828",
                "display": "flex",
                "fontFamily": "SFUIText-Semibold",
                "fontSize": 16,
                "fontStyle": "italic",
                "lineHeight": 24,
                "padding": 40,
              },
            ]
          }
        >
          Tap next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
